src/gui/advancedsettings.cpp

```diff
@@ -231,18 +231,18 @@ void AdvancedSettings::saveAdvancedSettings()
         session->setNetworkInterfaceAddress({});
     }
     else if (m_comboBoxInterfaceAddress.currentIndex() == 1) {
-        session->setNetworkInterfaceAddress({'4'});
+        session->setNetworkInterfaceAddress(QStringLiteral("0.0.0.0"));
     }
     else if (m_comboBoxInterfaceAddress.currentIndex() == 2) {
-        session->setNetworkInterfaceAddress({'6'});
+        session->setNetworkInterfaceAddress(QStringLiteral("::"));
     }
     else {
         QHostAddress ifaceAddr(m_comboBoxInterfaceAddress.currentText().trimmed());
-        ifaceAddr.isNull() ? session->setNetworkInterfaceAddress({}) : session->setNetworkInterfaceAddress(ifaceAddr.toString());
+        session->setNetworkInterfaceAddress(ifaceAddr.isNull() ? QString {} : ifaceAddr.toString());
```

Comment:
would this work: `session->setNetworkInterfaceAddress(ifaceAddr.isNull() ? {} : ifaceAddr.toString());`?

Reply:
Unfortunately no.
gcc:
```
error: expected primary-expression before '{' token
```
msvc:
```
syntax error: '{'
```